scripts/artifact-building/amphora-image/build_amphora_image.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+# Copyright 2014-2017 , Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Shell Opts ----------------------------------------------------------------
+
+set -e -u -x
+
+## Vars ----------------------------------------------------------------------
+
+# To provide flexibility in the jobs, we have the ability to set any
+# parameters that will be supplied on the ansible-playbook CLI.
+export ANSIBLE_PARAMETERS=${ANSIBLE_PARAMETERS:--v}
+
+# Set this to YES if you want to replace any existing artifacts for the current
+# release with those built in this job.
+export REPLACE_ARTIFACTS=${REPLACE_ARTIFACTS:-no}
+
+# Set this to YES if you want to push any changes made in this job to rpc-repo.
+export PUSH_TO_MIRROR=${PUSH_TO_MIRROR:-no}
+
+# The BASE_DIR needs to be set to ensure that the scripts
+# know it and use this checkout appropriately.
+export BASE_DIR=${PWD}
+
+# We want the role downloads to be done via git
+# This ensures that there is no race condition with the artifacts-git job
+export ANSIBLE_ROLE_FETCH_MODE="git-clone"
+
+## Main ----------------------------------------------------------------------
+
+# Bootstrap Ansible
+# This script is sourced to ensure that the common
+# functions and vars are available.
+source scripts/bootstrap-ansible.sh
+
+# Build an amphora image to be uplaoded
+# Todo: set directories properly
+openstack-ansible -i /opt/inventory \
+                  ${BASE_DIR}/scripts/artifacts-building/amphora-image/openstackgit-update.yml \
+                  -e rpc_release=${RPC_RELEASE} \
+                  ${ANSIBLE_PARAMETERS}
+
+# If there are artifacts for this release, then set PUSH_TO_MIRROR to NO
+if git_artifacts_available; then
```

Comment:
You'll need to replace this with a check of your own to determine whether an image already exists for the current release.